### ISSUE TITLE
changing host header does not work

### ISSUE THREAD

**Issue by MaorSabag** [BishopFox/sliver#1707]
**Describe the bug**
I have a frontdoor setup on azure for domain fronting (for example: frontdoor.azureedge.net), when generating an http/s payload and setting the host-header to the azure domain, it does not set it up when the payload is  executed.

**To Reproduce**
1. Generate http/s payload: 
```generate --http ajax.microsoft.com/api?driver=wininet&host-header=frontdoor.azureedge.net --skip-symbols```
3. Execute the payload (windows platform)
4. View the https traffic using proxy tool
5. the host header is not set  🥲

**Expected behavior**
the host header should be: frontdoor.azureedge.net with the domain of ajax.microsoft.com.

**Screenshots**
The request sent from the payload:
![image](https://github.com/BishopFox/sliver/assets/75939197/4547953d-9102-45c3-b087-e6d2896f039c)

Working request to the azure frontdoor but without setting the host header:
![image](https://github.com/BishopFox/sliver/assets/75939197/266d9d7d-169b-4510-b206-1a4875f6341a)

**Desktop (please complete the following information):**
 - OS: Payload running on Windows, sliver client and server are on Ubuntu.
 - Version v1.5.42

**Additional context**
Even tried to change the code manually (eventhough I'm so sure about), set the req.Host = "frontdoor.azureedge.net".
But it had not effect on the request.
![image](https://github.com/BishopFox/sliver/assets/75939197/0dc69ae4-8614-4a54-bc99-caccf7bf2cd0)

**Comment by navaneeth-dev** [comment]
Just tried to replicate it, it is working fine. Maybe try generating with -d and post the logs of the client?

**Comment by MaorSabag** [comment]
Screenshot with `-d` options:
![image](https://github.com/BishopFox/sliver/assets/75939197/505a3ce2-b15c-440a-bf48-03fde2922673)

**Comment by navaneeth-dev** [comment]
Oh I did not use wininet driver, can you try without that?

**Comment by MaorSabag** [comment]
Got a few insights 😅
Without the `driver` parameter set but with a proxy set on the Windows host, got the same results:
![image](https://github.com/BishopFox/sliver/assets/75939197/4cdb10d4-b97a-4978-9503-7d22b311d87d)

Without the `driver` parameter set and without the proxy set on the Windows host, got a successful result:
![image](https://github.com/BishopFox/sliver/assets/75939197/809e8d57-c05e-4291-a53e-7c4b1cb69bb5)


Seems like a bug.  I'm dealing with an organization proxy, thus the proxy options should be included if I want to use my frontdoor.

**Comment by navaneeth-dev** [comment]
Yes seems like a bug, I don't have the time right now but maybe others can look into it.

https://github.com/BishopFox/sliver/blob/3051de7a6152bf699f277a2aa71ba2d6c0a2d52b/implant/sliver/proxy/provider_windows.go#L287

I don't see any references to `WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY` which should detect the proxy automatically.
But I am not too sure.

**Comment by heyquentin** [comment]
> I had a very similar problem, but with Cloudflare instead of Azure. I solved it by removing the `driver=wininet` parameter when generating a beacon e.g.
> 
> ```
> generate beacon --http http://cdnjs.com?host-header=XXXXXXXX.worker.dev  --seconds 5 --jitter 4 --save /tmp/beacon_http2.exe
> ```

Do you have any ideas on how I'd do this using CloudFront?

**Comment by MaorSabag** [comment]
Hey, is there any update on this maybe?
